File: Java/lesson 15/AbsDistinct.java
// you can also use imports, for example:
import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 8
        Set<Integer> set = new HashSet<Integer>();
        for (int val: A) {
            if (val < 0) val *= -1;
            set.add(val);
        }
        return set.size();
    }
}

File: Go/lesson 15/AbsDistinct.go
package solution

// you can also use imports, for example:
// import "fmt"
// import "os"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

func Solution(A []int) int {
    // write your code in Go 1.4
    for i := 0; i < len(A); i++ {
        if A[i] < 0 {
            A[i] *= -1
        }
    }
    res := 0
    i, j := 0, len(A) - 1
    pre := -1
    for i <= j {
        if A[i] <= A[j] {
            if A[j] != pre {
                pre = A[j]
                res++
            }
            j--
        } else {
            if A[i] != pre {
                pre = A[i]
                res++
            }
            i++
        }
    }
    return res
}